File: leetcode/3110.rb
def score_of_string(s)
    def recur(cs, acc = 0)
        if cs.length < 2
            return acc
        end
        
        recur(cs[1..], acc + (cs[0].ord - cs[1].ord).abs)
    end

    recur(s.chars)
end

File: leetcode/3110.py
class Solution:
    def scoreOfString(self, s: str) -> int:
        def recur(o: str, acc: int = 0) -> int:
            if not o or len(o) < 2:
                return acc
            return recur(o[1:], acc + abs(ord(o[0]) - ord(o[1])))

        return recur(s)
